feat: add JSON-driven error catalog with compile-time validation

modcrafter77 opened the pull request:
Implement a type-safe error catalog system that reads error definitions from JSON at compile time and generates strongly-typed error handling code.

## New Error Catalog System

### Core Infrastructure
- Add modkit-errors: Runtime helpers for catalog-driven Problem building
  - ErrDef struct for static error definitions
  - 	o_problem() method to build Problems from catalog entries

- Add modkit-errors-macro: Proc-macro for compile-time code generation
  - declare_errors! macro reads JSON and generates error artifacts
  - Validates error codes, statuses, and titles at compile time
  - Auto-generates type URLs (https://errors.example.com/<code>)
  - Ensures Cargo rebuilds when JSON changes via include_str!

### Integration in users_info Module
- Add gts/errors.json with 8 error definitions
- Normalize validation errors to 422 status (invalid_email stays 400)
- Migrate error mapping to use catalog via ErrorCode::to_response()
- Remove hardcoded error codes, titles, and status values

## Additional ModKit Improvements (https://github.com/hypernetix/hyperspot/issues/15)

### Error Handling
- Refactor error_layer to manually attach instance/trace_id
- Remove finalize helpers in favor of direct builder pattern
- Improve Problem API with builder methods

### API & Runtime
- Enhance operation_builder with additional features
- Add runtime tests module with comprehensive test coverage
- Add capability suggestion for unknown macro capabilities

Artifizer commented on examples/modkit/users_info/gts/errors.json:
```diff
@@ -0,0 +1,42 @@
+  {
+    "status": 404,
+    "title": "User Not Found",
+    "code": "users_info.user.not_found.v1"
```
gts.hx.core.errors.err.v1~hx.example1.user.not_found.v1

Artifizer commented on examples/modkit/users_info/gts/errors.json:
```diff
@@ -0,0 +1,42 @@
+  {
+    "status": 409,
+    "title": "Email Already Exists",
+    "code": "users_info.user.email_conflict.v1"
```
gts.hx.core.errors.err.v1~hx.example1.user.email_conflict.v1

Artifizer commented on examples/modkit/users_info/gts/errors.json:
```diff
@@ -0,0 +1,42 @@
+  {
+    "status": 400,
+    "title": "Invalid Email",
+    "code": "users_info.user.invalid_email.v1"
```
gts.hx.core.errors.err.v1~hx.example1.user.invalid_email.v1

modcrafter77 committed:
feat: add JSON-driven error catalog with compile-time validation

Implement a type-safe error catalog system that reads error definitions from
JSON at compile time and generates strongly-typed error handling code.

## New Error Catalog System

### Core Infrastructure
- Add modkit-errors: Runtime helpers for catalog-driven Problem building
 - ErrDef struct for static error definitions
 - to_problem() method to build Problems from catalog entries
 - Extract Problem types to modkit-errors (pure data, no HTTP dependencies)

- Add modkit-errors-macro: Proc-macro for compile-time code generation
 - declare_errors! macro reads JSON and generates error artifacts
 - Validates error codes, statuses, and titles at compile time
 - Auto-generates type URLs (https://errors.example.com/<code>)
 - Ensures Cargo rebuilds when JSON changes via include_str!

- Rename odata-core → modkit-odata
  - Move OData/pagination logic to modkit-odata
  - Centralize OData error catalog
  - Update entire workspace to new structure

### Integration in users_info Module
- Add gts/errors.json with 8 error definitions
- Normalize validation errors to 422 status (invalid_email stays 400)
- Migrate error mapping to use catalog via ErrorCode::to_response()
- Remove hardcoded error codes, titles, and status values

## Additional ModKit Improvements (https://github.com/hypernetix/hyperspot/issues/15)

### Error Handling
- Refactor error_layer to manually attach instance/trace_id
- Remove finalize helpers in favor of direct builder pattern
- Improve Problem API with builder methods

### API & Runtime
- Enhance operation_builder with additional features
- Add runtime tests module with comprehensive test coverage
- Add capability suggestion for unknown macro capabilities
